script: Take away Fallible from new_resolved and new_rejected

kkoyung opened the pull request:
Both Promise::new_resolved and new_rejected only return `Ok`. We don't need them to be fallible. Simply return `Rc<Promise>`, instead of `Fallible<Rc<Promise>>`. Also, clean up relevant code.

<!-- Please describe your changes on the following line: -->


---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `___` with appropriate data: -->
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #35463 

<!-- Either: -->
- [x] These changes do not require tests because the returned types in tests also changes along with the code. 

<!-- Also, please make sure that "Allow edits from maintainers" checkbox is checked, so that we can help you if you get stuck somewhere along the way.-->

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

kkoyung committed:
script: Take away Fallible from new_resolved and new_rejected

Both Promise::new_resolved and new_rejected only return `Ok`. We don't
need them to be fallible. Simply return `Rc<Promise>`, instead of
`Fallible<Rc<Promise>>`. Also, clean up relevant code.

Signed-off-by: Kingsley Yung <kingsley@kkoyung.dev>

simonwuelker commented on components/script/dom/defaultteeunderlyingsource.rs:
```diff
@@ -115,11 +115,11 @@ impl DefaultTeeUnderlyingSource {
             let cx = GlobalScope::get_cx();
             rooted!(in(*cx) let mut rval = UndefinedValue());
-            return Some(Promise::new_resolved(
+            return Some(Ok(Promise::new_resolved(
```
It seems that this method is now also infallible

Gae24 replied:
Note that the Option can be also dropped here

kkoyung replied:
Okay. No problem.

simonwuelker commented on components/script/dom/readablestreamgenericreader.rs:
```diff
@@ -50,7 +50,7 @@ pub(crate) trait ReadableStreamGenericReader {
             // Set reader.[[closedPromise]] to a promise resolved with undefined.
             let cx = GlobalScope::get_cx();
-            self.set_closed_promise(Promise::new_resolved(global, cx, ())?);
+            self.set_closed_promise(Promise::new_resolved(global, cx, ()));
```
This method is now also infallible

kkoyung replied:
Great. I just did it in this way in the latest commit.

simonwuelker reviewed (changes requested):
Thanks, it's always nice to see fewer calls to `unwrap` in the code (:

There are a few other methods that become infallible due to this change, can you make the return a non-Result too?

kkoyung committed:
script: pull_algorithm becomes infallible

The method pull_algorithm only returns `Some(Ok(_))`, which means it is
infallible. Clean up the returned type.

Signed-off-by: Kingsley Yung <kingsley@kkoyung.dev>

kkoyung committed:
script: generic_initialize becomes infallible

The method generic_initialize only returns `Ok(())`, which means it is
infallible. Clean up the returned type.

Signed-off-by: Kingsley Yung <kingsley@kkoyung.dev>

jdm commented on components/script_bindings/codegen/CodegenRust.py:
```diff
@@ -807,13 +807,7 @@ def wrapObjectTemplate(templateBody, nullValue, isDefinitelyObject, type,
-                match Promise::new_resolved(&promiseGlobal, cx, valueToResolve.handle()) {
-                    Ok(value) => value,
-                    Err(error) => {
-                    throw_dom_exception(cx, &promiseGlobal, error);
```
If there's no error possible, we could never execute that code anyways!
